src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,34 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = EventBase.ScrubTenant(endpointUri),
+                QueryParams = String.Join("&", MsalHelpers.ParseKeyValueList(endpointUri.Query, '&', false, true, requestContext).Keys)
```

Comment:
I would still say move these telemetry specific code into the telemetry setters.